### DISCUSSION TITLE
Random error

### DISCUSSION THREAD

**Discussion by Aditi4AI** [mrdbourke/pytorch-deep-learning#508] [Q&A] [answered]
import torch
A=torch.rand(3,3)
B=A.type(torch.int32)
B
Can you explain me why this code👆 gives a output like this👇
tensor([[0, 0, 0],
[0, 0, 0],
[0, 0, 0]], dtype=torch.int32)

**Comment by Surya-Abhinai** [comment, marked as answer]
First, you are creating a tensor A of size 3x3 using the torch.rand function. By default, torch.rand generates a tensor with random values between 0 and 1 of the torch.float32 data type.

Next, you are converting the tensor A to the torch.int32 data type using the type method and assign it to a new tensor B. This conversion truncates the decimal parts of the values in A, rounding them down to the nearest integer.

As a result, the output tensor B will contain integer values. In this case, since the original tensor A was generated using torch.rand, which produces random values between 0 and 1, when converting to integers, all values will be rounded down to 0. Therefore, the output tensor B will consist of zeros.

Hope this helps .
For more information refer :- https://pytorch.org/docs/stable/generated/torch.rand.html

**Reply by Aditi4AI** [reply]
Thankyou very much!
